File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int a[] = new int[1001];
        List<Integer> ans = new ArrayList<>();
        for(int i=0;i<nums1.length;i++)
            a[nums1[i]]++;
        for(int i=0;i<nums2.length;i++) {
            if(a[nums2[i]]>0) {
                ans.add(nums2[i]);
                a[nums2[i]]--;
            }
        }
        a = new int[ans.size()];
        for(int i=0;i<ans.size();i++) {
            a[i] = ans.get(i);
        }
            
        return a;
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        int a[1001];
        vector<int> ans;
        int n1 = nums1.size();
        int n2 = nums2.size();
        for(int i=0;i<n1;i++)
            a[nums1[i]]++;
        for(int i=0;i<n2;i++)
            if(a[nums2[i]]>0) {
                ans.push_back(nums2[i]);
                a[nums2[i]]--;
            }
        return ans;
                
    }
};